### PR TITLE
Add dsl for SCM (now called VersionControl)

### PR DESCRIPTION
**Pull request by MasseGuillaume**
The scm url syntax is a source of confusion for developper. I added VersionControl.github() to simplify this process. We can add other common VersionControl url scheme like Git, Bazar, etc.

### CODE REVIEW

File: scalalib/src/mill/scalalib/publish/settings.scala [hunk 1]
```diff
@@ -58,11 +58,25 @@ case class License(
     distribution: String = "repo"
 )
 
-case class SCM(
+object SCM {
+  @deprecated("use VersionControl", "0.1.0")
```
Comment (rockjam) [inline scalalib/src/mill/scalalib/publish/settings.scala]: Not sure we should deprecate this

Comment (MasseGuillaume) [reply scalalib/src/mill/scalalib/publish/settings.scala]: I prefer VersionControl, it's self-explanatory.

**Comment by lihaoyi** [comment]
looks ok to me

File: docs/example-2/build.sc [hunk 2]
```diff
@@ -10,10 +10,7 @@ object foo extends PublishModule{
     organization = "com.lihaoyi",
     url = "https://github.com/lihaoyi/example",
     licenses = Seq(License.MIT),
-    scm = SCM(
-      "git://github.com/lihaoyi/example.git",
-      "scm:git://github.com/lihaoyi/example.git"
-    ),
+    versionControl = VersionControl.github("lihaoyi", "example"),
```
Comment (japgolly) [inline docs/example-2/build.sc]: Fantastic

**Pull request merged**